src/component/io.cpp

```diff
@@ -50,6 +50,28 @@ namespace io
 
 			return object;
 		}
+
+		scripting::script_value http_post(const gsc::function_args& args)
+		{
+			const auto url = args[0].as<std::string>();
+			const auto headers = args[1].as<utils::http::headers>();
```

Comment:
This is where linker error is happening mainly. Using `utils::http::headers` for some reason causes that error. I've even tried just using `std::unordered_map<std::string, std::string>` and it causes the same error.

Reply:
because there is no convertion from script_value to utils::http::headers or any std::unordered_map:
```
const auto array = args[0].as<scripting::array>();
const auto keys = array.get_keys();
utils::http::headers headers{};
for (const auto& key : keys)
{
	const auto value = array[key];
	if (!key.is<std::string>() || !value.is<std::string>())
	{
		continue;
	}

	const auto key_str = key.as<std::string>();
	headers[key_str] = value.as<std::string>();
}
```